frontend/src/pages/CourseSelector/courseDescription/AddToPlannerButton.jsx

```diff
@@ -0,0 +1,95 @@
+import { PlusOutlined, StopOutlined } from "@ant-design/icons";
+import { Button } from "antd";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addToUnplanned, removeCourse } from "../../../reducers/plannerSlice";
+
+const AddToPlannerButton = () => {
+  const { active, tabs } = useSelector((state) => state.courseTabs);
+  const coursesInPlanner = useSelector((state) => state.planner.courses);
+  const course = useSelector((state) => state.courses.course);
+
+  const id = tabs[active];
+  const dispatch = useDispatch();
+  const [addedCourseInPlanner, setAddedCourseInPlanner] = useState(!!coursesInPlanner[id]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (!!coursesInPlanner[id] !== addedCourseInPlanner) {
+      if (coursesInPlanner[id]) {
+        // course is being added
+        setLoading(true);
+        setTimeout(() => {
+          setLoading(false);
+          setAddedCourseInPlanner(true);
+        }, 1000);
+      } else if (!coursesInPlanner[id]) {
```

Comment:
these two clauses are the same in every respect except for setAddedCourseInPlanner - use a turnary statement to differentiate between the two